### ISSUE TITLE
More than new 300 "Duplicate code found" errors after updating patch version from v1.7.12 to v1.7.13

### ISSUE THREAD

**Issue by nickgnd** [rrrene/credo#1225]
Hello 👋
first of all, I want to thank you for this great library ❤️ 

---

In order to bump our elixir version from v1.18 to v.1.19 we needed to update Credo too, from v1.7.12 to v1.7.13.

After updating it, we went from 0 errors to 342 errors, all about code duplication. 

I cannot share the errors directly, but in some cases, the error reports the filename and the module name that have duplicated code, e.g.

```
┃ [D] ↑ Duplicate code found in lib/.../_template.ex:2 (mass: 188).
┃       lib/...other...model.ex:2 #(MyApp...OtherModel)
```

In other cases, the filename reported below the error is the same 🤔 

```
┃ [D] ↑ Duplicate code found in lib/mix/tasks/run_check.ex: (mass: 42).
┃       lib/mix/tasks/run_check.ex #()
```

Which is odd I think.

☝️ Moreover, when checking a few of the instances reported (when they have the filename), there is indeed duplicated code, but it should be acceptable imho (i.e. lower than threshold).

Then, comparing [v1.7.12 vs v1.7.13](https://github.com/rrrene/credo/compare/v1.7.12..v1.7.13) I only see this change for the `duplicated_code` rule, which I don't think is related 🤔 

<img width="1696" height="259" alt="Image" src="https://github.com/user-attachments/assets/00041700-450b-49c5-aced-0bde4b041ee3" />

Our credo config is pretty basic I'd say:

```elixir
%{
  configs: [
    %{
      name: "default",
      files: %{
        included: ["lib/", "test/", "mix.exs"],
        excluded: [~r"/_build/", ~r"/deps/", ~r"/node_modules/"]
      },
      plugins: [],
      requires: ["./lib/workera_credo/**/*.ex"],
      strict: true,
      parse_timeout: 5_000,
      color: true,
      checks: %{
        enabled: [
          {Credo.Check.Consistency.MultiAliasImportRequireUse, []},
          {Credo.Check.Design.DuplicatedCode, []},
          {Credo.Check.Readability.BlockPipe, []},
          {Credo.Check.Readability.AliasAs, []},
          {Credo.Check.Readability.ModuleDoc, []},
          {Credo.Check.Readability.MultiAlias, []},
          {Credo.Check.Readability.SingleFunctionToBlockPipe, []},
          {Credo.Check.Readability.SinglePipe, []},
          {Credo.Check.Readability.StrictModuleLayout, []},
          {Credo.Check.Readability.NestedFunctionCalls, [min_pipeline_length: 5]},
          {Credo.Check.Refactor.ABCSize, []},
          {Credo.Check.Refactor.AppendSingleItem, []},
          {Credo.Check.Refactor.DoubleBooleanNegation, []},
          {Credo.Check.Refactor.FilterReject, []},
          {Credo.Check.Refactor.IoPuts, []},
          {Credo.Check.Refactor.NegatedIsNil, []},
          {Credo.Check.Refactor.PipeChainStart, []},
          {Credo.Check.Refactor.RejectFilter, []},
          {Credo.Check.Refactor.VariableRebinding, []},
          {Credo.Check.Refactor.Nesting, [max_nesting: 4]},
          {Credo.Check.Warning.ApplicationConfigInModuleAttribute, []},
          {Credo.Check.Warning.Dbg, []},
          {Credo.Check.Warning.LeakyEnvironment, []},
          {Credo.Check.Warning.MapGetUnsafePass, []},
          {Credo.Check.Warning.MixEnv, []},
          {Credo.Check.Warning.IoInspect, []},
          {Credo.Check.Warning.UnsafeToAtom, []},
          {Credo.Check.Warning.WrongTestFileExtension, []},
          {WorkeraCredo.Check.Consistency.DependencySorting, []},
          {WorkeraCredo.Check.Consistency.TestModuleName, []},
          {WorkeraCredo.Check.Readability.PrivateFunctionBeforePublic, []}
        ],
        disabled: [
          {Credo.Check.Design.TagTODO, []},
          {Credo.Check.Readability.Specs, []},
          {Credo.Check.Readability.PredicateFunctionNames, []},
          {Credo.Check.Refactor.ModuleDependencies, []},
          {Credo.Check.Warning.MissedMetadataKeyInLoggerConfig, []}
        ]
      }
    }
  ]
}
```

I don't think is expected. Is it possible that there is a regression in Credo behaviour? Wdyt?

Thank you again for your time and let me know if I can help.
Cheers ✌️

**Comment by rrrene** [comment]
Thx for reporting this and the analysis 🙏

It seems this has to do with the switch from 1.18 to 1.19.

Running Credo 1.7.12 with Elixir 1.19 leads to an error.

Switching from Credo 1.7.12 with Elixir 1.18.4 to Credo 1.7.13 with Elixir 1.19.1 leads to lots of new issues, like you described.

**Comment by rrrene** [comment]
This should now be improved on `master`. 

I found a possible cause for the regression, but I am not sure that is causing all of the reported problems 🙏

Could you try this? Just set the Credo dep to

```elixir
{:credo, github: "rrrene/credo"}
```

and report back your results 🙂

**Comment by nickgnd** [comment]
Hey @rrrene thank you a lot for looking into that ❤️ 

I've just tested it and I can confirm that the problem is fixed. With the latest fix in master I don't have any Credo errors, which is consistent with Credo v1.7.12 and Elixir v1.18. 🚀 

Wonderful! Thank you again 👏

**Comment by epinault** [comment]
Hello! 

Duplicated code are gone for me too 

but now  I am seeing still new errors from master. Some more changes coming? or new checks?

```
[W] ↗ lib/crm/segments/postgres_query.ex:1413:5 length is expensive, prefer `Enum.empty?/1` or `list == []`.
```

any chance we can get the update soon then? would love to be able to upgrade to 1.19! thank you!

**Comment by nickgnd** [comment]
I tested the latest version v1.7.14 and I didn't get any new errors, thank again for looking into that 🙇

**Comment by ceolinrenato** [comment]
Working for me too, just updated to `v1.7.14` and brought back the duplicated code check into a project of mine

**Comment by rrrene** [comment]
@nickgnd @epinault @ceolinrenato Thank you for being so kind and sorry for the regression 🙏
